accepted/2020/feature-switch.md

```diff
@@ -0,0 +1,119 @@
+# Feature switch
+The functionality available in the .NET libraries is getting bigger and wider with every release. Applications typically don't need or want all of it, which means when an app includes the libraries as part of it, app size is unnecessarily increased. With technologies like the linker, we have the ability to remove parts of the framework which the app doesn't use. These technologies have limitations based on existing public API patterns and code behavior, sometimes they can't determine what the app really needs. This proposal adds a way to explicitly disable functionality in the framework.
+
+## Goals
+* Ability to declaratively turn-off functionality in the framework (and potentially any library/app)
+* Consistent behavior across different build configurations - the app should get the exact same behavior whether the linker was used on it or not
+* Ability to set different defaults based on the type of application (Blazor, Xamarin, …)
+
+
+## Expected usage
+### Size
+By disabling functionality which is otherwise always available the linker can trim more code from the app reducing its size.
+For example
+* Disable support for compiled regular expressions. The constructor for regular expression [`Regex(string, RegexOptions)`](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.-ctor?view=netcore-3.1#System_Text_RegularExpressions_Regex__ctor_System_String_System_Text_RegularExpressions_RegexOptions_) uses the second parameter to determine if it should pre-compile the regular expression or interpret it each time. The code ot pre-compile the expression is relatively large and if the app doesn't use this feature it can be removed. But linker typically won't be able to determine that since it's a dynamic behavior. (This is already [conditionally enabled](https://github.com/dotnet/runtime/blob/a83598dd151628672b1d2ff72310cc8c2065a019/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs#L77-L81) only when dynamic code compilation is supported by the runtime.)
```

Comment:
```suggestion
* Disable support for compiled regular expressions. The constructor for regular expression [`Regex(string, RegexOptions)`](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.-ctor?view=netcore-3.1#System_Text_RegularExpressions_Regex__ctor_System_String_System_Text_RegularExpressions_RegexOptions_) uses the second parameter to determine if it should pre-compile the regular expression or interpret it each time. The code to pre-compile the expression is relatively large and if the app doesn't use this feature it can be removed. But linker typically won't be able to determine that since it's a dynamic behavior. (This is already [conditionally enabled](https://github.com/dotnet/runtime/blob/a83598dd151628672b1d2ff72310cc8c2065a019/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs#L77-L81) only when dynamic code compilation is supported by the runtime.)
```